src/skydb.test.js

```diff
@@ -0,0 +1,154 @@
+  it("should set jsonData to skydb", async () => {
+    const receivedData = await client.db.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+describe("db.getJSON", () => {
+  it("should get jsonData from skydb", async () => {
+    const receivedData = await client.db.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("db.deleteJSON", () => {
+  it("should delete jsonData on skydb", async () => {
+    await client.db.deleteJSON(privateKey, dataKey);
+  });
+});
+
+describe("db.setJSON", () => {
+  it("This is important for the next tests with dbV2", async () => {
+    const receivedData = await client.db.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+//########################
+
+describe("dbV2.getJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("dbV2.setJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+describe("dbV2.getJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("dbV2.setDataLink", () => {
+  it("should be a DataLink set", async () => {
+    await client.dbV2.setDataLink(privateKey, dataKey, dataLink);
+  });
+});
+
+describe("dbV2.setEntryData", () => {
+  it("should set EntryData", async () => {
+    await client.dbV2.setEntryData(privateKey, dataKey, rawEntryData);
+  });
+});
+
+describe("dbV2.getEntryData", () => {
+  it("should get EntryData", async () => {
+    const receivedData = await client.dbV2.getEntryData(publicKey, dataKey);
+    await expect(receivedData["data"]).toEqual(rawEntryData);
+  });
+});
+
+describe("dbV2.getRawBytes", () => {
+  it("should get RawBytes", async () => {
+    const receivedData = await client.dbV2.getRawBytes(publicKey, dataKey);
+    await expect(receivedData["dataLink"]).toEqual(dataLink || dataLink_dbV2);
+    await expect("[" + receivedData["data"] + "]").toEqual(RawBytesData || RawBytesData_dbV2);
+  });
+});
+
+describe("dbV2.deleteJSON", () => {
+  it("should delete jsonData on skydb", async () => {
+    await client.dbV2.deleteJSON(privateKey, dataKey);
+  });
+});
+
+describe("dbV2.setJSON", () => {
+  it("This is important for the next tests with jest", async () => {
+    const receivedData = await client.db.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
```

Comment:
I don't understand what this test is for.

Reply:
Since all tests from skdb v1 and v2 are in one file and run one after the other, and the same variables are used for skydb v1 and v2, I had the problem when creating the test from dbV2 that it didn't work without existing jsonData.

That's why I set new data on Skydb with db.setJSON before the test dbV2 starts.

I can try again to leave this out. But I can not promise anything!
I think I can run it as a final test of skydb v1 but it would have the same effect.

Reply:
I would use different data for V1 and V2, if possible, or even random data. But if that's too much work you can just leave it as-is - I know you put a lot of time into this code, already.